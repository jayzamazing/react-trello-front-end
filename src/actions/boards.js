import {normalize} from 'normalizr';
import {boardArray, boardsSchema} from '../board-schema';
import {BASE_URL} from '../config';

/*
* action to tell store that all boards has been retrieved
* @params boards - boards to be sent to store
* @returns action type and boards
*/
export const FIND_BOARDS_SUCCESS = 'FIND_BOARDS_SUCCESS';
export const findBoardsSuccess = boards => {
  //grab all boards array with cards and cardslist and normalize it
  const items = (normalize(boards, boardArray)).entities;
  return {
    type: FIND_BOARDS_SUCCESS,
    items
  };
};
/*
* function to get all boards
* @params findBoardsSuccess or passed in action
* @dispatch findBoardsSuccess or passed in action
*/
export const getBoards = (action = findBoardsSuccess) => dispatch => {
  return fetch(`${BASE_URL}/boards`)
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(res.body));
    });
};
/*
* action to tell store that a board has been created
* @params boards - boards to be sent to store
* @returns action type and boards
*/
export const CREATE_BOARD_SUCCESS = 'CREATE_BOARD_SUCCESS';
export const createBoardSuccess = boards => {
  const items = (normalize(boards, boardsSchema)).entities;
  return {
    type: CREATE_BOARD_SUCCESS,
    items
  };
};
/*
* function to create a board
* @params postData - boards to update on the board
* @params createBoardSuccess or passed in action
* @dispatch createBoardSuccess or passed in action
*/
export const createBoards = (postData, action = createBoardSuccess) => dispatch => {
  return fetch(`${BASE_URL}/boards`, {
    method: "POST",
    body: postData
  })
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(res.body));
    });
};
/*
* action to tell store that a board has been deleted
* @params id - id of the board to delete in the store
* @returns action type and boards
*/
export const DELETE_BOARD_SUCCESS = 'DELETE_BOARD_SUCCESS';
export const deleteBoardSuccess = id => {
  return {
    type: DELETE_BOARD_SUCCESS,
    boardId: id
  };
};
/*
* function to delete a board
* @params id - id of the board to delete
* @params deleteBoardSuccess or passed in action
* @dispatch deleteBoardSuccess or passed in action
*/
export const deleteBoards = (id, action = deleteBoardSuccess) => dispatch => {
  return fetch(`${BASE_URL}/boards/${id}`, {
    method: 'DELETE'
  })
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(id));
    });
};
/*
* action to tell store that a board has been updated
* @params boards - boards to be sent to store
* @params id - id of the board to update in the store
* @returns action type and boards
*/
export const UPDATE_BOARD_SUCCESS = 'UPDATE_BOARD_SUCCESS';
export const updateBoardSuccess = function(id, boards) {
  const items = (normalize(boards, boardsSchema)).entities;
  return {
    type: UPDATE_BOARD_SUCCESS,
    items,
    boardId: id
  };
};
/*
* function to update the board
* @params id - id of the board to update
* @params postData - boards to update on the board
* @params updateBoardSuccess or passed in action
* @dispatch updateBoardSuccess or passed in action
*/
export const updateBoards = (id, postData, action = updateBoardSuccess) => dispatch => {
  return fetch(`${BASE_URL}/boards/${id}`, {
    method: 'PUT',
    body: postData
  })
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(id, postData));
    });
};
