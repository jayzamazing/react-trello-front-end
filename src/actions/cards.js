import {normalize} from 'normalizr';
import {cardsSchema} from '../board-schema';
import {BASE_URL} from '../config';
/*
* action to tell store that a cards has been created
* @params data - data to be sent to store
* @returns action type and data
*/
export const CREATE_CARDS_SUCCESS = 'CREATE_CARDS_SUCCESS';
export const createCardsSuccess = cards => {
  const items = (normalize(cards, cardsSchema)).entities;
  return {
    type: CREATE_CARDS_SUCCESS,
    items
  };
};
/*
* function to create a cards
* @params postData - data to update on the cards
* @params createCardsSuccess or passed in action
* @dispatch createCardsSuccess or passed in action
*/
export const createCards = (postData, action = createCardsSuccess) => dispatch => {
  return fetch(`${BASE_URL}/cards`, {
    method: "POST",
    body: postData
  })
    .then(res => {
      if (!res.ok) return Promise.reject(res.statusText);
      return dispatch(action(res.body));
    });
};
/*
* action to tell store that a cards has been created
* @params data - data to be sent to store
* @returns action type and data
*/
export const DELETE_CARDS_SUCCESS = 'DELETE_CARDS_SUCCESS';
export const deleteCardsSuccess = id => {
  return {
    type: DELETE_CARDS_SUCCESS,
    cardsId: id
  };
};
/*
* function to delete a cards
* @params id - id of the cards to delete
* @params deleteCardsSuccess or passed in action
* @dispatch deleteCardsSuccess or passed in action
*/
export const deleteCards = (id, action = deleteCardsSuccess) => dispatch => {
  return fetch(`${BASE_URL}/cards/${id}`, {
    method: 'DELETE'
  })
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(id));
    });
};
/*
* action to tell store that a cards has been updated
* @params cards - cards to be sent to store
* @params id - id of the cards to update in the store
* @returns action type and cards
*/
export const UPDATE_CARDS_SUCCESS = 'UPDATE_CARDS_SUCCESS';
export const updateCardsSuccess = function(id, cards) {
  const items = (normalize(cards, cardsSchema)).entities;
  return {
    type: UPDATE_CARDS_SUCCESS,
    items,
    cardsId: id
  };
};
/*
* function to update the cards
* @params id - id of the cards to update
* @params postData - cards to update on the board
* @params updateCardsSuccess or passed in action
* @dispatch updateCardsSuccess or passed in action
*/
export const updateCards = (id, postData, action = updateCardsSuccess) => dispatch => {
  return fetch(`${BASE_URL}/cards/${id}`, {
    method: 'PUT',
    body: postData
  })
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(id, postData));
    });
};
