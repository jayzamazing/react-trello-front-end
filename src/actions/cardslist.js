'use strict';
import {normalize} from 'normalizr';
import {cardsListSchema} from '../board-schema';

/*
* action to tell store that a cardslist has been created
* @params data - data to be sent to store
* @returns action type and data
*/
export const CREATE_CARDSLIST_SUCCESS = 'CREATE_CARDSLIST_SUCCESS';
export const createCardslistSuccess = cardslist => {
  const items = (normalize(cardslist, cardsListSchema)).entities;
  return {
    type: CREATE_CARDSLIST_SUCCESS,
    items
  };
};
/*
* function to create a cardslist
* @params postData - data to update on the cardslist
* @params createCardslistSuccess or passed in action
* @dispatch createCardslistSuccess or passed in action
*/
export const createCardslist = (postData, action = createCardslistSuccess) => dispatch => {
  return fetch('/cardslist', {
    method: "POST",
    body: postData
  })
    .then(res => {
      if (!res.ok) return Promise.reject(res.statusText);
      return dispatch(action(res.body));
    });
};
/*
* action to tell store that a cardslist has been created
* @params data - data to be sent to store
* @returns action type and data
*/
export const DELETE_CARDSLIST_SUCCESS = 'DELETE_CARDSLIST_SUCCESS';
export const deleteCardslistSuccess = id => {
  return {
    type: DELETE_CARDSLIST_SUCCESS,
    cardslistId: id
  };
};
/*
* function to delete a cardslist
* @params id - id of the cardslist to delete
* @params deleteCardslistSuccess or passed in action
* @dispatch deleteCardslistSuccess or passed in action
*/
export const deleteCardslist = (id, action = deleteCardslistSuccess) => dispatch => {
  return fetch(`/cardslist/${id}`, {
    method: 'DELETE'
  })
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(id));
    });
};
/*
* action to tell store that a cardslist has been updated
* @params cardslist - cardslist to be sent to store
* @params id - id of the cardslist to update in the store
* @returns action type and cardslist
*/
export const UPDATE_CARDSLIST_SUCCESS = 'UPDATE_CARDSLIST_SUCCESS';
export const updateCardslistSuccess = function(id, cardslist) {
  const items = (normalize(cardslist, cardsListSchema)).entities;
  return {
    type: UPDATE_CARDSLIST_SUCCESS,
    items,
    cardslistId: id
  };
};
/*
* function to update the cardslist
* @params id - id of the cardslist to update
* @params postData - cardslist to update on the board
* @params updateCardslistSuccess or passed in action
* @dispatch updateCardslistSuccess or passed in action
*/
export const updateCardslist = (id, postData, action = updateCardslistSuccess) => dispatch => {
  return fetch(`/cardslist/${id}`, {
    method: 'PUT',
    body: postData
  })
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(id, postData));
    });
};
