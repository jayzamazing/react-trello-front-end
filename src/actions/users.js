import {SubmissionError} from 'redux-form';
import {BASE_URL} from '../config';
import {normalizeResponseErrors} from './utils';

export const createUser = user => dispatch => {
  return fetch(`${BASE_URL}/users`, {
    method: 'POST',
    headers: {
      'content-type': 'application/json'
    },
    body: JSON.stringify(user)
  }).then(res => normalizeResponseErrors(res))
  .then(res => res.json())
  .catch(err => {
    const {reason, message, location} = err;
    return Promise.reject(
      return Promise.reject(
        new SubmissionError({
          [location]: message
        });;
      );
    );
  });
}
