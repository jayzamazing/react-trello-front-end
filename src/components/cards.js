var React = require('react');
var connect = require('react-redux').connect;
var actions = require('./CardsActions');
import CreateItems from '../utils/create-items';
import {Immutable} from 'seamless-immutable';
//component to store list of cards and text
var Cards = React.createClass({
  //set up initial data state
  getInitialState: function() {
    return {
      showCreateCards: false,
      editCards: {},
      cards: {},
      text: ''
    };
  },
  //keep track of text
  onAddInputChanged: function(event) {
    //if the addCards input is being used
    if (event.target.name == 'addCards') {
      this.setState({text: event.target.value});
    //otherwise assume we are editing cards name
    } else {
      //get cards from state
      var temp = this.state.cards;
      //update the title for the selected cards
      var temp2 = Immutable.update(temp,
      event.target.id,
      function() {
        return {
          text: event.target.value
        };
      });
      //store the updated cards
      this.setState({cardslist: temp2});
    }
  },
  //function to add a new card
  addCards: function() {
    // this.props.dispatch(
    //   //dispatch query cards
    //   actions.queries('cards', 'POST', {text: this.state.text},
    //   'create cards', this.props.cardslistId, this.props.boardId)
    // );
    //hide the following input
    this.setState({showCreateCards: false});
  },
  //function to delete a card
  deleteCards: function(cardId) {
    //dispatch query cards
    // this.props.dispatch(
    //   actions.queries('cards', 'DELETE', cardId, 'delete cards')
    // );
  },
  //function to edit the text in a card
  updateCards: function(cardsId, cardsText) {
    // this.props.dispatch(
    //   actions.queries('cards', 'PUT', {text: cardsText}, 'update cards', cardsId)
    // );
    this.forceUpdate();
  },
  showCreateCards: function() {
    this.setState({showCreateCards: true});
  },
  //deal with the user hitting enter from the input and updating the cards
  handleKeyPress: function(events) {
    if(events.charCode == 13) {
      var temp = this.state.editCards;
      temp[events.target.id] = true;
      this.setState({editCards: temp});
      this.updateCards(events.target.id, events.target.value);
    }
  },
  //set variable to enable the editing of the cards text
  editCardsText: function(item) {
    var temp = this.state.editCards;
    temp[item] = false;
    this.setState({editCards: temp});
  },
  render: function() {
    var context = this;
    var cardslist = context.props.cardslist[Object.keys(context.props.cardslist).find(item => {
        //if the id of props.cardslist matches cardslistid
      return context.props.cardslist[item]._id == context.props.cardslistId;
    })];
    //function to render multiple cards
    var cards = Object.keys(this.props.cards).map((item, index) => {
      if (cardslist.cards.indexOf(this.props.cards[item]._id) > -1) {
        var temp = context.props.cards[item];
        return (
          <li key={index}>
            <input type="text" id={temp._id} value={context.state.cards[temp._id] ? context.state.cards[temp._id].text : temp.text}
              disabled={(context.state.editCards[temp._id] == undefined) ? true : context.state.editCards[temp._id]}
              onChange={context.onAddInputChanged}
              onKeyPress={context.handleKeyPress}/>
            <input type="button" value="Delete Card"
              onClick={context.deleteCards.bind(null, temp._id)}/>
              <input type="button" value="Edit Card"
                onClick={context.editCardsText.bind(null, temp._id)}/>
          </li>
        );
      }
    });
    return (
      <div>
        <ul>
          {cards}
        </ul>
          <input type="button" value="Add Cards" onClick={this.showCreateCards}/>
          {this.state.showCreateCards ? <CreateItems
              onAddInputChanged={this.onAddInputChanged}
              addItems={this.addCards} name="addCards"/> : null}
      </div>
    );
  }
});
//allows subcription to redux updates and access to data stored in redux store
var mapStateToProps = function(state) {
  return {
    cardslist: state.cardslist,
    cards: state.cards
  };
};
var Container = connect(mapStateToProps)(Cards);
module.exports = {
  Container,
  Cards
};
