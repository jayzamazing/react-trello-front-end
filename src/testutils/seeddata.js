'use strict';
import faker from 'faker';
//used to create a single board in json
export const seedBoardsSingle = (title = faker.random.words(), _id = faker.random.alphaNumeric(20)) => {
  return {
    title: title,
    _id: _id,
    cardsList: seedCardslistIds()
  };
};
//used to create random boards based on count
export const seedBoardsCount = (count) => {
  if (!count) {
    count = 1;
  }
  let boards = [];
  for (var i = 0; i < count; i++) {
    let _id = faker.random.alphaNumeric(20);
    boards.push({
      _id: _id,
      title: faker.random.words(),
      cardsList: seedCardslistIds(count)
    });
  }
  return boards;
}
//create board items based on input values, after normalization
export const seedBoardsInput = (ids, words, ids2) => {
  let boards = {};
  for (var i = 0; i < ids.length; i++) {
    var _id = ids[i];
    boards[_id] = {
      _id: _id,
      title: words[i],
      cardslist: ids2[i]
    };
  }
  return boards;
}
//used to create cardslist ids for seedboards
export const seedCardslistIds = (count) => {
  let cardslistIds = [];
  for (var i = 0; i < count; i++) {
    cardslistIds.push(faker.random.alphaNumeric(20));
  }
  return cardslistIds;
}
//used to create a single cardslist in json
export const seedCardslistSingle = (boardId = faker.random.alphaNumeric(20), title = faker.random.words(), _id = faker.random.alphaNumeric(20)) => {
  return {
    boardId: boardId,
    title: title,
    _id: _id,
    cards: seedCardsIds()
  };
}
//used to create random cardslist based on count
export const seedCardslistCount = (boardId, count) => {
  if (!count) {
    count = 1;
  }
  let cardslist = [];
  for (var i = 0; i < count; i++) {
    cardslist.push({
      boardId: boardId,
      _id: faker.random.alphaNumeric(20),
      title: faker.random.words(),
      cards: seedCardsIds(count)
    });
  }
  return cardslist;
}
//create cardslist items based on input value, after normalization
export const seedCardslistInput = function(boardId, words, ids) {
  let cardslist = {};
  for (var i = 0; i < ids.length; i++) {
    var _id = ids[i];
    cardslist[_id]= {
      boardId: boardId[i],
      _id: _id,
      title: words[i],
      cards: seedCardsIds(ids.length)
    };
  }
  return cardslist;
};
//used to create cardslist ids for seedboards
export const seedCardsIds = (count) => {
  let cardsIds = [];
  for (var i = 0; i < count; i++) {
    cardsIds.push(faker.random.alphaNumeric(20));
  }
  return cardsIds;
}
//used to create a single card in json
export const seedCardsSingle = (cardslistId = faker.random.alphaNumeric(20), text = faker.random.words(), _id = faker.random.alphaNumeric(20)) => {
  return {
    cardslistId: cardslistId,
    text: text,
    _id: _id
  };
}
//used to create random cardslist based on count
export const seedCardsCount = (cardslistId, count) => {
  if (!count) {
    count = 1;
  }
  let cards = [];
  for (var i = 0; i < count; i++) {
    cards.push({
      cardslistId: cardslistId,
      _id: faker.random.alphaNumeric(20),
      text: faker.random.words()
    });
  }
  return cards;
}
//create cards items based on input value, after normalization
export const seedCardsInput = function(cardslistId, words, ids) {
  let cards = {};
  for (var i = 0; i < ids.length; i++) {
    var _id = faker.random.alphaNumeric(20);
    cards[_id]= {
      cardslistId: cardslistId[i],
      _id: _id,
      text: faker.random.words()
    };
  }
  return cards;
};
//create a random title
export const createTitle = () => {
return {title: faker.random.words()};
};
