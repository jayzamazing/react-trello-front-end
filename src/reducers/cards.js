import Immutable from 'seamless-immutable';
import {FIND_BOARDS_SUCCESS} from '../actions/boards';
import {CREATE_CARDS_SUCCESS, DELETE_CARDS_SUCCESS, UPDATE_CARDS_SUCCESS} from '../actions/cards';

const initialRepositoryState = Immutable({});

/*
* Function called when DELETE_CARDS_SUCCESS is called to deal with deleting a Cards and removing it from state
* params state- old state before merge
* params action.cardsid- Cards to delete
*/
export const deleteCards= (state, {cardsId}) => state.without(cardsId);

/*
* Function called when UPDATE_CARDS_SUCCESS is called to update cardslsit with a new cards
* params state- old state before merge
* params action.items.cardslsit- cards to update
*/
export const updateCards= (state, {cards}) => state.merge(cards, {deep: true});

/*
* Function to deal with using various reducer functions
* params state- old state before merge
* params action- action with data to modify state
*/
export default (state = initialRepositoryState, action) => {
  switch (action.type) {
  case FIND_BOARDS_SUCCESS:
  case CREATE_CARDS_SUCCESS:
  case UPDATE_CARDS_SUCCESS:
    if (action.cards) {
      return updateCards(state, action);
    } else {
      return state;
    }
  case DELETE_CARDS_SUCCESS:
    return deleteCards(state, action);
  default:
    return state;
  }
};
