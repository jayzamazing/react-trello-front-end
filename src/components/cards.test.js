'use strict';
import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow, mount} from 'enzyme';
import {Cards} from './cards';
import {seedCardslistInput, seedCardsInput} from '../testutils/seeddata';
import * as actions from '../actions/cards';
import renderer from 'react-test-renderer';

//mock actions
const mockDeleteCards = {
  type: 'DELETE_CARDS'
};
const mockUpdateCards = {
  type: 'UPDATE_CARDS'
};
//setup mock api
jest.mock('../actions/cards', () => Object.assign({},
require.requireActual('../actions/cards'),
{
  //needed for true comparison
  deleteCards: jest.fn().mockImplementation(() => {
    return mockDeleteCards;
  }),
  //needed for true comparison
  updateCards: jest.fn().mockImplementation(() => {
    return mockUpdateCards;
  })
}
));
//testing for cards component
describe('Cards component', function() {
  let cards = {};
  let cardslist = {};
  //setup before all tests
  beforeEach(() => {
    boardId = '46mdujckqs45tkdsh2jdxz';
    cardsListId = '2skdde39hzdo08aqcdgd';
    let ids = ['2skdde39hzdo08aqcdgd', '14w43d4opubteu021ato', 'kd5o3ve2858vpjxd7wjp'];
    let words = ['XML Health', 'Bedfordshire Illinois Liaison', 'Sports Analyst Cambridgeshire'];
    let ids2 = ['432e39hzdo08aqcdgd', '6532opubteu021ato', '5643e2858vpjxd7wjp'];
    let words2 = ['eat dinner', 'tea tree oil', 'masking tape'];
    cardslist = seedCardslistInput('46mdujckqs45tkdsh2jdxz', ids, words);
    cards = seedCardsInput('2skdde39hzdo08aqcdgd', ids2, words2);
  });
  //tear down after tests are complete
  afterEach(() => {
    cards = {};
    cardslist = {};
  });
  //ensure card is rendered without issues
  it('should render', () => {
    const dispatch = jest.fn();
    shallow(<Cards cards={cards} boardId={boardId}
    cardsList={cardsList} cardsListId={cardsListId}/>);
  });
  // it('Renders the Cards item', function() {
  //   //create instance of render
  //   var renderer = TestUtils.createRenderer();
  //   //render an image component
  //   renderer.render(<Cards.CardsContainer cards={cardItems.cards} boardId={cardItems.boardId}
  //     cardsList={cardItems.cardsList} cardsListId={cardItems.cardsListId}/>);
  //   //get the rendered react component to test against
  //   var result = renderer.getRenderOutput();
  //   //test props for various values
  //   result.type.should.equal('div');
  //   var resultListItem = result.props.children;
  //   resultListItem[0].type.should.equal('ul');
  //   var cards = resultListItem[0].props.children;
  //   // console.log(result);
  //   cards[0].props.children[0].props.value.should.equal(cardItems.cards[1].text);
  // });
  //test for performing click event on add cards
  // it('should simulate a click event on add Cards input', () => {
  //   //set up a mockstore
  //   const store = mockStore({
  //     cardsList: cardItems.cardsList,
  //     cards: cardItems.cards
  //   });
  //   //create instance of render and pass store to it
  //   let renderer = TestUtils.renderIntoDocument(
  //     <Provider store={store}>
  //       <Cards.Container boardId={cardItems.boardId} cardsListId={cardItems.cardsListId}/>
  //     </Provider>
  //   );
  //   //get the input for cards
  //   let inputs = TestUtils.scryRenderedDOMComponentsWithTag(renderer, 'input');
  //   inputs.length.should.equal(4);
  //   TestUtils.Simulate.click(inputs[3]);
  //   let inputs2 = TestUtils.scryRenderedDOMComponentsWithTag(renderer, 'input');
  //   // //check that previous input is there plus two inputs from create-items
  //   inputs2.length.should.equal(6);
  //   inputs2[4].value = 'happy';
  //   TestUtils.Simulate.change(inputs2[4]);
  //   //simulate button click
  //   TestUtils.Simulate.click(inputs2[5]);
  // });
  //test for deleting a card
  // it('should simulate a click event on delete cardslist input', () => {
  //   //set up a mockstore
  //   const store = mockStore({
  //     cardsList: cardItems.cardsList,
  //     cards: cardItems.cards
  //   });
  //   //create instance of render and pass store to it
  //   let renderer = TestUtils.renderIntoDocument(
  //     <Provider store={store}>
  //       <Cards.Container boardId={cardItems.boardId} cardsListId={cardItems.cardsListId}/>
  //     </Provider>
  //   );
  //   //get the input for boards
  //   let inputs = TestUtils.scryRenderedDOMComponentsWithTag(renderer, 'input');
  //   inputs.length.should.equal(4);
  //   //simulate button click
  //   TestUtils.Simulate.click(inputs[1]);
  // });
  // it('should simulate a click event on edit cards input', () => {
  //   //set up a mockstore
  //   const store = mockStore({
  //     cardsList: cardItems.cardsList,
  //     cards: cardItems.cards
  //   });
  //   //create instance of render and pass store to it
  //   let renderer = TestUtils.renderIntoDocument(
  //     <Provider store={store}>
  //       <Cards.Container boardId={cardItems.boardId} cardsListId={cardItems.cardsListId}/>
  //     </Provider>
  //   );
  //   //get the input for boards
  //   let inputs = TestUtils.scryRenderedDOMComponentsWithTag(renderer, 'input');
  //   inputs.length.should.equal(4);
  //   //simulate button click
  //   TestUtils.Simulate.click(inputs[2]);
  //   inputs[3].value = 'happy';
  //   TestUtils.Simulate.change(inputs[0]);
  //   //simulate button click
  //   TestUtils.Simulate.keyDown(inputs[0], {key: 'Enter', keyCode: 13, which: 13});
  // });
});
