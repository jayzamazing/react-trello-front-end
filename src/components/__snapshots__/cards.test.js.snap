// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards component add cards click snaphot 1`] = `
<div
  className="cards-form"
>
  <ul
    className="cards-list-ul"
  >
    <li
      key="0"
    >
      <div
        className="cards-tile"
      >
        <div
          className="update-card"
        >
          <UpdateCards
            _id="432e39hzdo08aqcdgd"
            card={
              Object {
                "title-0": undefined,
              }
            }
            cardslist={
              Object {
                "14w43d4opubteu021ato": Object {
                  "_id": "14w43d4opubteu021ato",
                  "boardId": "346rfdghsdhteu021ato",
                  "cards": Array [
                    "6532opubteu021ato",
                  ],
                  "title": "Bedfordshire Illinois Liaison",
                },
                "2skdde39hzdo08aqcdgd": Object {
                  "_id": "2skdde39hzdo08aqcdgd",
                  "boardId": "46mdujckqs45tkdsh2jdxz",
                  "cards": Array [
                    "432e39hzdo08aqcdgd",
                  ],
                  "title": "XML Health",
                },
                "kd5o3ve2858vpjxd7wjp": Object {
                  "_id": "kd5o3ve2858vpjxd7wjp",
                  "boardId": "kd5o3ve2858qggfqw4rtfda",
                  "cards": Array [
                    "5643e2858vpjxd7wjp",
                  ],
                  "title": "Sports Analyst Cambridgeshire",
                },
              }
            }
            index={0}
          />
        </div>
      </div>
    </li>
    <li>
      <div
        className="create-cards"
      >
        <div
          className="create-cards-link"
        >
          <span
            className="createcards"
            id="createcards"
            name="createcards"
            onClick={[Function]}
            value="Add Cards"
          >
            Add a card...
          </span>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`Cards component add cards click snaphot 2`] = `
<div
  className="cards-form"
>
  <ul
    className="cards-list-ul"
  >
    <li
      key="0"
    >
      <div
        className="cards-tile"
      >
        <div
          className="update-card"
        >
          <UpdateCards
            _id="432e39hzdo08aqcdgd"
            card={
              Object {
                "title-0": undefined,
              }
            }
            cardslist={
              Object {
                "14w43d4opubteu021ato": Object {
                  "_id": "14w43d4opubteu021ato",
                  "boardId": "346rfdghsdhteu021ato",
                  "cards": Array [
                    "6532opubteu021ato",
                  ],
                  "title": "Bedfordshire Illinois Liaison",
                },
                "2skdde39hzdo08aqcdgd": Object {
                  "_id": "2skdde39hzdo08aqcdgd",
                  "boardId": "46mdujckqs45tkdsh2jdxz",
                  "cards": Array [
                    "432e39hzdo08aqcdgd",
                  ],
                  "title": "XML Health",
                },
                "kd5o3ve2858vpjxd7wjp": Object {
                  "_id": "kd5o3ve2858vpjxd7wjp",
                  "boardId": "kd5o3ve2858qggfqw4rtfda",
                  "cards": Array [
                    "5643e2858vpjxd7wjp",
                  ],
                  "title": "Sports Analyst Cambridgeshire",
                },
              }
            }
            index={0}
          />
        </div>
      </div>
    </li>
    <li>
      <div
        className="create-cards"
      >
        <ReduxForm
          cardslist={
            Object {
              "_id": "2skdde39hzdo08aqcdgd",
              "boardId": "46mdujckqs45tkdsh2jdxz",
              "cards": Array [
                "432e39hzdo08aqcdgd",
              ],
              "title": "XML Health",
            }
          }
          close={[Function]}
          onSubmit={[Function]}
        />
      </div>
    </li>
  </ul>
</div>
`;

exports[`Cards component cards snapshot 1`] = `
<div
  className="cards-form"
>
  <ul
    className="cards-list-ul"
  >
    <li
      key="0"
    >
      <div
        className="cards-tile"
      >
        <div
          className="update-card"
        >
          <UpdateCards
            _id="432e39hzdo08aqcdgd"
            card={
              Object {
                "title-0": undefined,
              }
            }
            cardslist={
              Object {
                "14w43d4opubteu021ato": Object {
                  "_id": "14w43d4opubteu021ato",
                  "boardId": "346rfdghsdhteu021ato",
                  "cards": Array [
                    "6532opubteu021ato",
                  ],
                  "title": "Bedfordshire Illinois Liaison",
                },
                "2skdde39hzdo08aqcdgd": Object {
                  "_id": "2skdde39hzdo08aqcdgd",
                  "boardId": "46mdujckqs45tkdsh2jdxz",
                  "cards": Array [
                    "432e39hzdo08aqcdgd",
                  ],
                  "title": "XML Health",
                },
                "kd5o3ve2858vpjxd7wjp": Object {
                  "_id": "kd5o3ve2858vpjxd7wjp",
                  "boardId": "kd5o3ve2858qggfqw4rtfda",
                  "cards": Array [
                    "5643e2858vpjxd7wjp",
                  ],
                  "title": "Sports Analyst Cambridgeshire",
                },
              }
            }
            index={0}
          />
        </div>
      </div>
    </li>
    <li>
      <div
        className="create-cards"
      >
        <div
          className="create-cards-link"
        >
          <span
            className="createcards"
            id="createcards"
            name="createcards"
            onClick={[Function]}
            value="Add Cards"
          >
            Add a card...
          </span>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
